src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -256,22 +250,30 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                 // CRLF for end of headers.
                 await WriteTwoBytesAsync((byte)'\r', (byte)'\n', cancellationToken).ConfigureAwait(false);
 
-                // Write body, if any
+                // Flush out anything buffered from the request headers to finish sending them.
+                await FlushAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
> Don't buffer content writes

I was wondering about that, too, and actually briefly changed it not to and then undid it thinking it was too much for this PR.  But I'll redo it.

> if the initial content write can fit in the remaining part of the write buffer, then copy it into the buffer so we can do a single write

Yup

> Obviously, if there's no content, we need to flush immediately

Yup

> If the user cares about sending the request headers without sending any content, they can explicitly do a flush.

Ok.

> aren't you going to want to send something in the body, to start things off?

It may very well be unlikely.  I'm just thinking of analogies where it's not, e.g. you make a phone call to an automated system, and you as the client don't say anything but instead listen to the response content "please select from one of the following options... press 1 for blah, press 2 for blah..." and only after getting the response content do you then send request content.  But, that's just an analogy... I don't know if it actually manifests in this realm.